### ISSUE TITLE
Inconsistency in Human3.6 Dataset splits 

### ISSUE THREAD

**Issue by matteorr** [hrhodin/UnsupervisedGeometryAwareRepresentationLearning#5]
Hi, thanks a lot for your great work!

I have a question on the exact data splits used. In Fig.4.a and in Sec.5.1 you mention that the largest training set used is composed by the pose annotations of all 5 subjects (S1, S5, S6, S7, S8) sampled at 10fps (so effectively you take one pose every 5 frames since the original videos are at 50fps - please correct me if wrong) for a total of about 370000 examples.

However, I am not able to recreate splits with the same number of annotations. When loading all the poses I get the following numbers:

- S1: 248376 total frames (~49K when divided by 5)
- S1+S5: 644692 total frames (~129K when divided by 5)
- S1+S5+S6: 894556 total frames (~179K when divided by 5)
- S1+S5+S6+S7+S8: 1559752 total frames **(~310K when divided by 5)**

As you can see, the last split has around 60K less annotations than the value reported. In particular:

- I am using the h5 annotations from [this repo](https://github.com/una-dinosauria/3d-pose-baseline), which ones are you using?
- What are the exact numbers you get for each split?
- Could you please help me understand what might be the cause of the inconsistency?

Thanks a lot!

**Comment by matteorr** [comment]
Hi @hrhodin, just wondering if you had a chance to look into this! Thanks a lot!

**Comment by hrhodin** [comment]
Hi matteorr,

Thanks for checking this inconsistency. It is a typo. As you noted, all other subsets match exactly, only the "all subject" case is off.
I just loaded that dataset again, it gives me 312 038 frames, which matches with your estimate.

Best, Helge

**Comment by hrhodin** [comment]
Here the exact numbers for different number of subjects (subject ids in braces):
[1]:49 702
[1,5]: 128 984
[1,5,6]: 178 966
[1,5,6,7]: 260 284
[1,5,6,7,8]: 312 038

**Comment by matteorr** [comment]
Hi Helge, thanks for your thorough and helpful reply! A quick follow-up, so I can compare my results more precisely. 

1. Using S9 and S11 and considering the video that is missing (S11, Directions, Camera 54138969) I have 550644 test frames (~110K after down-sampling). Could you please confirm that the test set you report results on in the paper is the same dimensions?

2. Do you perform any data augmentation on the input images (i.e. cropping, rescaling, color)? If so, could you please point me to where I could find it in the code so I can replicate the exact setup?

Thanks!

**Comment by matteorr** [comment]
Hi Helge - @hrhodin, did you see my follow-up? Any help would be great. Thanks!

**Comment by hrhodin** [comment]
1. I used the dataset from the paper Towards 3D Human Pose Estimation in the Wild: a Weakly-supervised Approach, that was kindly provided by Xingyi Zhou
Apparently, the validation set (S9 and S11) are subsampled to every 200th frame, starting with the first one. This amounts to 2814 frames.
2. As briefly explained in the paper, we perform rotation augmentation (in-plane rotation, with the angle sampled from a triangular distribution from -20 to +20 degrees) and perspective correction.
No color, scale, not translation augmentation is done.

**Comment by Ixiaohuihuihui** [comment]
Could you  tell me how to  get the file "labels.h5" ? And what is the data structure? Thank you very much!

**Comment by hrhodin** [comment]
Please send me a mail for details on the dataset processing.

**Comment by matteorr** [comment]
Hi @hrhodin , I have another quick follow-up and was hoping you could help me! 

When training the self-supervised geometry aware representation do you maintain the same training/test split of the supervised part or, (since it is not using ground-truth 3d pose labels) do you use videos from all the subjects of human3.6m?

Thanks a lot!

**Comment by hrhodin** [comment]
All training subjects are used for unsupervised training, usually larger than the supervised subset.

**Comment by vru2020** [comment]
Hi @matteorr: can you please share the dataset processing you received from @hrhodin ? Thanks a lot

**Comment by hrhodin** [comment]
Please send me a mail for details on the dataset processing.
